syntax = "proto3";

package pb_test;

message Person {
  string name = 1;
  uint32 id = 2;
  string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }
  // repeated field should use a smaller tag so that the message is smaller.
  repeated float scores = 5 [packed=true];
  message PhoneNumber {
    uint32 number = 1;
    PhoneType type = 2;
  }
  
  repeated PhoneNumber phones = 4;  
  string address = 8;
  string nationality = 9;
  string birthday = 10;
  string creationDate = 11;
  string updateDate = 12;
  string selfIntroduction = 13;
}

message University {
  repeated Person students = 1;
}
